File: articles/grit-and-self-control-related-but-not-the-same/fortran/grit_self_control_score.f90
program grit_self_control_score
  implicit none
  real :: attention_regulation, emotion_regulation, impulse_control
  real :: perseverance_effort, consistency_interests
  real :: self_control, grit

  attention_regulation = 0.74
  emotion_regulation = 0.62
  impulse_control = 0.81
  perseverance_effort = 0.91
  consistency_interests = 0.68

  self_control = 0.40 * attention_regulation + 0.30 * emotion_regulation + 0.30 * impulse_control
  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests

  print *, "Synthetic self-control score:", self_control
  print *, "Synthetic grit score:", grit
end program grit_self_control_score

File: articles/grit-and-self-control-related-but-not-the-same/cpp/grit_self_control_score.cpp
#include <iostream>

double self_control_score(double attention_regulation, double emotion_regulation, double impulse_control) {
    return 0.40 * attention_regulation + 0.30 * emotion_regulation + 0.30 * impulse_control;
}

double grit_score(double perseverance_effort, double consistency_interests) {
    return 0.60 * perseverance_effort + 0.40 * consistency_interests;
}

int main() {
    double sc = self_control_score(0.74, 0.62, 0.81);
    double g = grit_score(0.91, 0.68);

    std::cout << "Synthetic self-control score: " << sc << "\n";
    std::cout << "Synthetic grit score: " << g << "\n";
    return 0;
}
